crates/core/app/tests/common/test_node_builder_ext.rs

```diff
@@ -56,7 +56,7 @@ impl BuilderExt for Builder {
 /// Generates a [`Validator`][PenumbraValidator] given a consensus verification key.
 pub fn generate_penumbra_validator(
     consensus_key: &ed25519_consensus::VerificationKey,
-) -> (PenumbraValidator, Allocation) {
+) -> (PenumbraValidator, Allocation, penumbra_keys::keys::SpendKey) {
```

Comment:
We don't want to return the spend key here, since this method shouldn't be used in any situation where we need access to the spend key.

This method should be purely internal to the `with_penumbra_auto_app_state` method -- where we discard the validator's signing keys post-setup -- and not exposed for external use.

Reply:
i cherry-picked ceb5d62 into this branch.

Reply:
3ca8c462e